cassandra/tablets.py

```diff
@@ -51,7 +58,42 @@ def get_tablet_for_key(self, keyspace, table, t):
             return tablet[id]
         return None
 
+    def drop_tablet(self, keyspace: str, table: str = None):
+        print("drop tablets for ", keyspace, table)
+        with self._lock:
+            if table is not None:
+                self._tablets.pop((keyspace, table), None)
+                return
+
+            to_be_deleted = []
+            for key in self._tablets.keys():
+                if key[0] == keyspace:
+                    to_be_deleted.append(key)
+
+            for key in to_be_deleted:
+                del self._tablets[key]
+
+    def drop_tablet_by_host_id(self, host_id: UUID):
+        print("drop tablets for ", host_id)
+        if host_id is None:
+            return
+        with self._lock:
+            for key, tablets in self._tablets.items():
+                to_be_deleted = []
+                for tablet_id, tablet in enumerate(tablets):
+                    if tablet.replica_contains_host_id(host_id):
+                        to_be_deleted.append(tablet_id)
+
+                if len(to_be_deleted) == 0:
+                    continue
+
+                for tablet_id in reversed(to_be_deleted):
+                    print("drop tablets - ", tablets.pop(tablet_id))
+
+                self._tablets[key] = tablets
+
```

Comment:
🔧 I don't see why either `if len(to_be_deleted) == 0:` or `self._tablets[key] = tablets`

`self._tablets[key] = tablets` is a no-op - you mutate `tablets` in-place, so you just assign the same object that is already there.

`if len(to_be_deleted) == 0:` is not necessary - without it we will skip the loop, and then execute the no-op assingment.

Reply:
fixed